src/parameterizations/vertical/MOM_kappa_shear.F90

```diff
@@ -441,10 +531,63 @@ subroutine Calc_kappa_shear_vertex(u_in, v_in, h, T_in, S_in, tv, p_surf, kappa_
   ! Convert layer thicknesses into geometric thickness in height units.
   call thickness_to_dz(h, tv, dz_3d, G, GV, US, halo_size=1)
 
+  if (CS%VS_ThicknessMean) then
+    hweight_ul(:,:,:) = 0.0
+    hweight_ur(:,:,:) = 0.0
+    hweight_ll(:,:,:) = 0.0
+    hweight_lr(:,:,:) = 0.0
+    h_int_mask(:,:,:) = 0.0
+    !Loop over halo and pre compute a masked thickness around each interface
+    ! (excluding the top and bottom boundary)
+    do j=G%jsc-1,G%jec+1; do K=2,nz; do i=G%isc-1,G%iec+1
+      h_int_mask(i,j,K) = G%mask2dT(i,j)*(h(i,j,k-1)+h(i,j,k))/2.
+    enddo; enddo ; enddo
+    !Compute the thickness weighting if requested for horizontal mean
+    do J=JsB,JeB ; do K=2,nz ; do I=IsB,IeB
+      hweight_denom = h_int_mask(i-1,j-1,k) + &   !left,bot
+                      2.*h_int_mask(i-1,j,k) + &  !left,cen
+                      h_int_mask(i-1,j+1,k) + &   !left,top
+                      2. * h_int_mask(i,j-1,k) + &!mid,bot
+                      4. * h_int_mask(i,j,k) + &  !mid,cen
+                      2. * h_int_mask(i,j+1,k) + &!mid,top
+                      h_int_mask(i+1,j-1,k) + &   !right,bot
+                      2. * h_int_mask(i-1,j,k) + &!right,cen
+                      h_int_mask(i+1,j+1,k) + &   !right,top
+                      GV%H_subroundoff
+      ! Computing the weighting of the 4 vertices to each cell center
+      hweight_ul(I,J,K) = ((h_int_mask(i-1,j,k) + &   !left,cen
+                            h_int_mask(i-1,j+1,k)) + &!left,top
+                           (h_int_mask(i,j,k) + &     !mid,cen
+                            h_int_mask(i,j+1,k)) &    !mid,top
+                           ) / hweight_denom
+      hweight_ur(I,J,k) = ((h_int_mask(i,j,k) + &   !mid,cen
+                            h_int_mask(i,j+1,k)) + &!mid,top
+                           (h_int_mask(i+1,j,k) + & !right,cen
+                            h_int_mask(i+1,j+1,k)) &!right,top
+                            ) / hweight_denom
+      hweight_ll(I,J,k) = ((h_int_mask(i-1,j-1,k) + &!left,bot
+                            h_int_mask(i-1,j,k)) + & !left,cen
+                           (h_int_mask(i+1,j-1,k) + &!mid,bot
+                            h_int_mask(i+1,j,k)) &   !mid,cen
+                            ) / hweight_denom
+      hweight_lr(I,J,k) = ((h_int_mask(i,j-1,k) + &  !mid,bot
+                            h_int_mask(i,j,k)) + &   !mid,cen
+                           (h_int_mask(i+1,j-1,k) + &!right,bot
+                            h_int_mask(i+1,j,k)) &   !right,cen
+                           ) / hweight_denom
+    enddo ; enddo ; enddo
+  else
+    ! Set some weights for horizontal averaging factors
+    weight_ll=0.25
```

Comment:
These array-syntax expressions should be written as `weight_ll(:,:,:) = 0.25` to emphasize what they are doing, or better yet do not use these 3-d arrays when they are not needed, as this will unnecessarily reduce the code efficiency.